pygmt/datasets/tile_map.py

```diff
@@ -16,6 +16,11 @@
     TileProvider = None
     _HAS_CONTEXTILY = False
 
+try:  # noqa: SIM105
+    import rioxarray  # rioxarray is needed to register the rio accessor  # noqa: F401
+except ImportError:
+    pass
```

Comment:
This seems to work for me with `ruff=0.5.1`, maybe the `noqa: F401` comment is needed? Also need to add `import contextlib` at the top.

```suggestion
with contextlib.suppress(ImportError):
    import rioxarray  # noqa: F401
```